src/software/simulated_tests/ball_moves_in_direction.py

```diff
@@ -0,0 +1,135 @@
+import software.python_bindings as tbots
+from proto.import_all_protos import *
+
+from software.simulated_tests.validation import (
+    Validation,
+    create_validation_geometry,
+    create_validation_types,
+)
+
+
+class BallMovesXAxis(Validation):
+    """Checks if ball is moving forward or backward in the +x direction"""
+
+    def __init__(self, initial_ball_position: tbots.Point, moving_in_pos_x: bool):
+        """
+        Constructs the validation
+        :param initial_ball_position: the initial position of the ball
+        :param moving_in_pos_x: True if checking for +x direction, false if checking for -x direction
+        """
+        self.last_ball_position = initial_ball_position
+        self.moving_in_pos_x = moving_in_pos_x
```

Comment:
do we need this parameter? Isn't the always not version of this test handling this?